src/PostgREST/DbStructure.hs

```diff
@@ -621,63 +583,44 @@ allPrimaryKeys tabs =
     -- CTE to replace information_schema.table_constraints to remove owner limit
     */
     WITH tc AS (
-        SELECT current_database()::information_schema.sql_identifier AS constraint_catalog,
-            nc.nspname::information_schema.sql_identifier AS constraint_schema,
-            c.conname::information_schema.sql_identifier AS constraint_name,
-            current_database()::information_schema.sql_identifier AS table_catalog,
-            nr.nspname::information_schema.sql_identifier AS table_schema,
-            r.relname::information_schema.sql_identifier AS table_name,
+        SELECT current_database()::name AS constraint_catalog,
+            nc.nspname::name AS constraint_schema,
+            c.conname::name AS constraint_name,
+            current_database()::name AS table_catalog,
+            nr.nspname::name AS table_schema,
+            r.relname::name AS table_name,
                 CASE c.contype
-                    WHEN 'c'::"char" THEN 'CHECK'::text
-                    WHEN 'f'::"char" THEN 'FOREIGN KEY'::text
-                    WHEN 'p'::"char" THEN 'PRIMARY KEY'::text
-                    WHEN 'u'::"char" THEN 'UNIQUE'::text
-                    ELSE NULL::text
-                END::information_schema.character_data AS constraint_type,
-                CASE
-                    WHEN c.condeferrable THEN 'YES'::text
-                    ELSE 'NO'::text
-                END::information_schema.yes_or_no AS is_deferrable,
-                CASE
-                    WHEN c.condeferred THEN 'YES'::text
-                    ELSE 'NO'::text
-                END::information_schema.yes_or_no AS initially_deferred
+                    WHEN 'c' THEN 'CHECK'
+                    WHEN 'f' THEN 'FOREIGN KEY'
+                    WHEN 'p' THEN 'PRIMARY KEY'
+                    WHEN 'u' THEN 'UNIQUE'
+                    ELSE NULL
+                END::text AS constraint_type
         FROM pg_namespace nc,
             pg_namespace nr,
             pg_constraint c,
             pg_class r
-        WHERE nc.oid = c.connamespace AND nr.oid = r.relnamespace AND c.conrelid = r.oid AND (c.contype <> ALL (ARRAY['t'::"char", 'x'::"char"])) AND r.relkind = 'r'::"char" AND NOT pg_is_other_temp_schema(nr.oid)
-        /*--AND (pg_has_role(r.relowner, 'USAGE'::text) OR has_table_privilege(r.oid, 'INSERT, UPDATE, DELETE, TRUNCATE, REFERENCES, TRIGGER'::text) OR has_any_column_privilege(r.oid, 'INSERT, UPDATE, REFERENCES'::text))*/
-        UNION ALL
-        SELECT current_database()::information_schema.sql_identifier AS constraint_catalog,
```

Comment:
This part of the union can never be used, as the `contraint_type` is hardcoded to be 'CHECK' and we filter for 'PRIMARY_KEY' in the final query (`tc.constraint_type = 'PRIMARY KEY'`)

Reply:
Yes! I can see why it's unnecessary here since we are only looking for pk constraints.